diesel_tests/tests/update.rs

```diff
@@ -307,3 +307,30 @@ fn upsert_with_sql_literal_for_target() {
     ];
     assert_eq!(Ok(expected_data), data);
 }
+
+#[test]
+#[cfg(feature = "postgres")]
+fn update_array_index_expression() {
+    use crate::schema::posts::dsl::*;
+
+    let connection = &mut connection_with_sean_and_tess_in_users_table();
+
+    let sean = find_user_by_name("Sean", connection);
+    let new_post = sean.new_post("Hello", Some("world"));
+    insert_into(posts)
+        .values(&new_post)
+        .execute(connection)
+        .unwrap();
+    update(posts)
+        .set(tags.eq(vec!["programming", "rust"]))
+        .execute(connection)
+        .unwrap();
+    update(posts)
+        .set(tags.index(1).eq("postgres"))
+        .execute(connection)
+        .unwrap();
```

Comment:
Apologies that I can't contribute much to this review, but just want to understand that these two update statements are unblocked by this PR? If I understand correctly right now `diesel` doesn't provide any way to update records in the DB based on literals and this PR allows it to happen? If so, very neat!

If that's too reductive, then the only feedback I might be is to add tests for other types of updates (maybe a few from the PG documentation like an operator invocation (a + b), type cast (lit as i32), etc.)